mitodl/micromasters
# Add mitX Online as an alternative social auth backend

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #4989 

#### What's this PR do?
This PR adds mitx online as an alternative social auth backend. It does not expose this in the UI yet.

#### How should this be manually tested?
Follow the [setup steps](https://github.com/mitodl/micromasters/pull/4996/files#diff-d64c15ff6ea885e0d24eed19dd5c1db82759fffa0f15c43f029ddd8853882fba)

Verify the following authentication flows work:

- Can register a new "edx.org" account
- Can register a new mitx online account
- Can add mitx online account to an existing "edx.org" account
- For the following account types, verify you can still login:
  - an account with only an "edx.org" social auth
  - an account with only an mitxonline social auth
  - an account with both kinds of social auth (verify that you can sign in with both platforms)
